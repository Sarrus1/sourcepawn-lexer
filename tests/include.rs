mod utils;

use insta::assert_json_snapshot;
use sourcepawn_lexer::*;
use utils::collect_tokens;

#[test]
fn include_simple_1() {
    let input = r#"#include <sourcemod>
int foo;"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn opening_chevron_1() {
    let input = r#"1 < 2"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn include_line_continuation_1() {
    let input = r#"#include <sourcemod\
>"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}
