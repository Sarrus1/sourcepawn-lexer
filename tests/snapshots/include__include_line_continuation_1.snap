---
source: tests/include.rs
assertion_line: 30
expression: collect_tokens(&mut lexer)
snapshot_kind: text
---
[
  {
    "kind": {
      "PreprocDir": "MInclude"
    },
    "text": "#include <sourcemod\\\n>",
    "range_start": 0,
    "range_end": 22,
    "delta": 0,
    "in_preprocessor": true
  },
  {
    "kind": "Eof",
    "text": "\u0000",
    "range_start": 22,
    "range_end": 22,
    "delta": 0,
    "in_preprocessor": false
  }
]
